### PR TITLE
feat: display created by user in search

### PR DESCRIPTION
**Pull request by kwasniew**
## About the changes

Displaying columns with created by user:
Screenshot 2024-06-05 at 17 40 54

Currently we show avatar and user name + id on hover.

Next PR:
* extract and reuse on hover component from the project overview
* add filter by user id backend (new filter in search and list of all users who created features in this project) and UI component

## Discussion points

Should we put it behind a dedicated flag? It's a new table column that's visible by default on large screens.

### CODE REVIEW

**Comment by ivarconr** [comment]
Have we considered how it looks for old feature flags created before we supported created_by_user_id? Will it then just say system user?

**Comment by ivarconr** [comment]
> Should we put it behind a dedicated flag? It's a new table column that's visible by default on large screens.

Whats the argument to not flag this?

File: frontend/src/openapi/models/featureSearchResponseSchema.ts [hunk 1]
```diff
@@ -28,7 +28,7 @@ export interface FeatureSearchResponseSchema {
      */
     createdAt: string | null;
     /** User who created the feature flag */
-    createdBy?: FeatureSearchResponseSchemaCreatedBy;
+    createdBy: FeatureSearchResponseSchemaCreatedBy;
```
Comment (thomasheartman) [inline frontend/src/openapi/models/featureSearchResponseSchema.ts:31]: will this always have a created by even in old flags? We did do some sort of patching of pre-existing flags, didn't we?

File: src/lib/features/feature-search/feature-search-store.ts [hunk 2]
```diff
@@ -404,6 +405,11 @@ class FeatureSearchStore implements IFeatureSearchStore {
 
             if (!entry) {
                 // Create a new entry
+                const name =
+                    row.user_name ||
+                    row.user_username ||
+                    row.user_email ||
+                    'unknown';
```
Comment (thomasheartman) [inline src/lib/features/feature-search/feature-search-store.ts:412]: Is the unknown case where we don't know who created the flag initially? Like Ivar asked, what do we show then?

**Review by thomasheartman — APPROVED** [review]
Looks fine to me, but I've got the same questions Ivar has. And like him, I'd lean towards flagging this initially. Probably not a huge deal, but gives us the option to turn it off if something goes wrong (reversible decisions 😉 )

File: src/lib/features/feature-search/feature-search-store.ts (hunk 2 above)
Comment (kwasniew) [reply src/lib/features/feature-search/feature-search-store.ts:412]: We show this
Screenshot 2024-06-06 at 09 42 21

**Pull request merged**